frontend/tests/emailWorkflow.test.ts

```diff
@@ -152,3 +153,107 @@ test('register via new-user invitation email', async ({ page, mailboxFactory })
   // Should be able to open sena-3 project from user dashboard as we are now a member
   await userDashboardPage.openProject('Sena 3', 'sena-3');
 });
+
+test('ask to join project via new-project page', async ({ page, tempUser, tempUserInTestOrg }) => {
+    test.setTimeout(TEST_TIMEOUT_2X);
+
+    // First, set up new user to be manager of Elawa project (since it doesn't have one in default seed data)
+    const manager = tempUser;
+    await loginAs(page.request, manager.email, manager.password);
+    let dashboardPage = await new UserDashboardPage(page).goto();
+
+    // Must verify email before being made manager of a project
+    await dashboardPage.emailVerificationAlert.assertPleaseVerify();
+    let newPage = await verifyTempUserEmail(page, manager);
+
+    // Add manager to Elawa project
+    await loginAs(newPage.request, 'admin');
+    await addUserToProject(newPage.request, manager.id, elawaProjectId, 'MANAGER');
+
+    const { name, email, password } = tempUserInTestOrg;
+
+    await loginAs(page.request, email, password);
+    dashboardPage = await new UserDashboardPage(page).goto();
+
+    // Must verify email before being allowed to request project creation
+    await dashboardPage.emailVerificationAlert.assertPleaseVerify();
+    newPage = await verifyTempUserEmail(page, tempUserInTestOrg);
+    dashboardPage = await new UserDashboardPage(newPage).goto();
+
+    // Create project with similar name to Elawa
+    const newProjectPage = await dashboardPage.clickCreateProject();
+    await newProjectPage.fillForm({name: 'Elaw', code: 'xyz', purpose: 'Testing', organization: 'Test Org'});
+    await expect(newProjectPage.extraProjectsDiv).toBeVisible();
+    await expect(newProjectPage.askToJoinBtn).toBeDisabled();
+    await newProjectPage.extraProjectsDiv.locator('#extra-projects-elawa-dev-flex').check();
```

Comment:
can we make this a helper function on `CreateProjectPage`? something like `relatedProjectItem(projectCode: string): Locator` just because we've got a bit of magic strings here with the current locator.

Reply:
Done in commit 531bd7c5.